### DISCUSSION TITLE
How to extract textbox text?

### DISCUSSION THREAD

**Discussion by jiezhoujenny** [pymupdf/PyMuPDF#776] [Q&A] [answered]
Hi, 
Is there any method that I can extract all the text in text box in a pdf file page? as attached picture shows (in red rectangle).
Thank you!
![image](https://user-images.githubusercontent.com/17275074/102658531-bc24a880-4145-11eb-9790-9d49377354a2.png)

**Comment by JorjMcKie** [comment, marked as answer]
The method is `page.getTextbox(rect)`.
But it seems in your example, some text pieces are not page text, but annotation or widget / field text. That text would of course **_not_** be included, because annotations (including fields / widgets) are not part of the page. You can imagine these like "dust" on a picture - can be wiped away without chaging the picture itself.

But you can still extract annotation / field contents, just with another set of methods:
You have to extract annot / field text separately, by first checking which fields / annotations are inside the given rect, and then extracting the text from them.

**Reply by JorjMcKie** [reply]
E.g. for annotations:
```python
for annot in page.annots():
    if annot.rect not in rect:
        continue
    print(annot.getTextbox())
```
For fields, you can use the field value instead:
```python
for field in page.widgets():
    if field.rect not in rect:
        continue
    print(field.field_value)
```

**Comment by jiezhoujenny** [comment]
Awesome! Thank you very much!
